src/v/cluster/cluster_discovery.cc

```diff
@@ -26,6 +26,19 @@ using std::vector;
 
 namespace cluster {
 
+namespace {
+
+void verify_duplicate_seed_servers() {
+    std::vector<config::seed_server> s = config::node().seed_servers();
+    std::sort(s.begin(), s.end());
+    const auto s_dupe_i = std::adjacent_find(s.cbegin(), s.cend());
+    if (s_dupe_i != s.cend())
+        throw std::runtime_error(fmt_with_ctx(
+          fmt::format, "Duplicate items in seed_servers: {}", *s_dupe_i));
+}
+
+} // namespace
+
```

Comment:
For me the main purpose of this commit is to validate the seed servers list. This is not reflected in the commit message